lib/ansible/galaxy/role.py

```diff
@@ -379,7 +379,29 @@ def install(self):
                                     if n_part != '..' and not n_part.startswith('~') and '$' not in n_part:
                                         n_final_parts.append(n_part)
                                 member.name = os.path.join(*n_final_parts)
-                                role_tar_file.extract(member, to_native(self.path))
+
+                                # Implemented the following code to circumvent broken implementation of data_filter
+                                # in tarfile. See for more information - https://github.com/python/cpython/issues/107845
+                                # deprecated: description='probing broken data filter implementation' python_version='3.12'
```

Comment:
Since 3.12.0 hasn't been released yet, we should be able to safely use 3.11 here. I don't think we need to worry about supporting 3.12 pre-releases 2+ years from now:

```suggestion
                                # deprecated: description='probing broken data filter implementation' python_version='3.11'
```